### PR TITLE
Adds support for readOnly masterKey

### PR DESCRIPTION
**Pull request by flovilmart**

### CODE REVIEW

File: Parse-Dashboard/app.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ module.exports = function(config, options) {
     // Serve the configuration.
     app.get('/parse-dashboard-config.json', function(req, res) {
       let response = {
-        apps: config.apps,
+        apps: [...config.apps], // make a copy
```
Comment (acinader) [inline Parse-Dashboard/app.js]: neat trick.

File: Parse-Dashboard/app.js [hunk 2]
```diff
@@ -101,14 +101,29 @@ module.exports = function(config, options) {
...
+      }
 
       if (successfulAuth) {
         if (appsUserHasAccess) {
           // Restric access to apps defined in user dictionary
           // If they didn't supply any app id, user will access all apps
           response.apps = response.apps.filter(function (app) {
             return appsUserHasAccess.find(appUserHasAccess => {
-              return app.appId == appUserHasAccess.appId
+              const isSame = app.appId == appUserHasAccess.appId;
```
Comment (acinader) [inline Parse-Dashboard/app.js]: prefer === :).  i know its not yours....

File: Parse-Dashboard/Authentication.js [hunk 3]
```diff
@@ -81,6 +81,7 @@ function initialize(app, options) {
 function authenticate(userToTest, usernameOnly) {
   var appsUserHasAccessTo = null;
   var matchingUsername = null;
+  var isReadOnly = false;
```
Comment (acinader) [inline Parse-Dashboard/Authentication.js]: switch all these to `let`?  your call :)....

File: Parse-Dashboard/Authentication.js [hunk 4]
```diff
@@ -96,12 +97,14 @@ function authenticate(userToTest, usernameOnly) {
         matchingUsername = user.user;
         // User restricted apps
         appsUserHasAccessTo = user.apps || null;
+        isReadOnly = user.readOnly;
       }
 
       return isAuthenticated;
     }) ? true : false;
```
Comment (acinader) [inline Parse-Dashboard/Authentication.js]: ewww, remove the `? true: false`? as completely redundant?

File: Parse-Dashboard/Authentication.js (hunk 3 above)
Comment (flovilmart) [reply Parse-Dashboard/Authentication.js]: yep good call

File: Parse-Dashboard/Authentication.js (hunk 4 above)
Comment (flovilmart) [reply Parse-Dashboard/Authentication.js]: yeap

File: Parse-Dashboard/Authentication.js [hunk 5]
```diff
@@ -96,15 +97,17 @@ function authenticate(userToTest, usernameOnly) {
         matchingUsername = user.user;
         // User restricted apps
         appsUserHasAccessTo = user.apps || null;
+        isReadOnly = !!user.readOnly; // make it true/false
       }
 
       return isAuthenticated;
-    }) ? true : false;
+    }) ? true : false ;
```
Comment (acinader) [inline Parse-Dashboard/Authentication.js]: mistake?

Comment (flovilmart) [reply Parse-Dashboard/Authentication.js]: no because of the && && it was returning an object instead of true / false

File: README.md [hunk 6]
```diff
@@ -282,6 +282,63 @@ When `user1` logs in, he/she will be able to manage `myAppId1` and `myAppId2` fr
...
 When *`user2`*  logs in, he/she will only be able to manage *`myAppId1`* from the dashboard.
 
+## Use Read-Only masterKey
+
+Starting parse-server 2.6.5, it is possible to provide a readOnlyMasterKey to parse-server to prevent mutations on objects from a client.
+If you want to protect your dashboard with this feature, just use the readOnlyMasterKey instead of the masterKey. All write calls will fail.
+
+### Making an app read-only for all users
+
+Start your parse server with
```
Comment (acinader) [inline README.md]: parse-server for consistency.

Comment (flovilmart) [reply README.md]: Yes!

Comment (acinader) [reply README.md]: also *could* put `readOnlyMasterKey` `masterKey` insteaxc of masterKey....i.e. mark as code.

Comment (flovilmart) [reply README.md]: yes
!

**Pull request merged**